### ISSUE TITLE
uncaught error when .sqrt() for a large number

### ISSUE THREAD

**Issue by n4o847** [MikeMcl/big.js#107]
When I try this
```
Big("1e1000").minus(1).sqrt()
```
I get
```
Uncaught Error: [big.js] Invalid number
```
(I think this is because in big.js#L775, "Infinity" is passed as the argument to Big.)

**Comment by MikeMcl** [comment]
The number is invalid because it has too many integer digits for the `sqrt` method to handle. This is an undocumented limitation.

The method could be extended to handle numbers with more integer digits than the current limit of about 308, see [bignumber.js](https://github.com/MikeMcl/bignumber.js/blob/master/bignumber.js#L2315), but this is a minimalist library and such numbers are impractically large anyway.

Thanks for the report. I will consider whether to extend the method accordingly, or at least document the limitation.

**Comment by n4o847** [comment]
I see. Certainly that is worth considering.

I might use bignumber.js for the time being. Thanks.

**Comment by MikeMcl** [comment]
Fixed in *v5.2.0*.
